students/KennanY/Lesson5/MailRoomPt3.py

```diff
@@ -0,0 +1,137 @@
+from datetime import datetime
+import os
+
+flag=True
+
+#Data structures
+people={'Jeff Bezos': [1005.49, 3116.72, 5200],
+             'Kelly Hirsch': [21.47, 1500],
+             'Alice Wong': [2400.54],
+             'Malcolm Wang': [355.42, 579.31],
+             'Kayla McAllison': [636.9, 850.13, 125.23],
+             'Jack Boyle': [66.9, 85.13, 443.23]}
+
+#Functions
+def printemail(name, amount):
+    ''' Print Email message'''
+    print("\nDear " + name +'\n')
+    print('We would like to thank you for continuing support for our organization!\n')
+    print('Your contribution of $' + str(amount) + ' is received and very much appreciated!')
+    print('\nThank you\nKennan Yilmaz')
+
+def generate_letters(donationdata):
+    for donor, amount in donationdata.items():
+        #Build file name
+        dt= str(datetime.now().strftime('%Y%m%d%H%M%S'))
+        filename=donor + dt + ".txt"
+
+        #Write to file
+    try:
+        with open(filename, 'w') as file:
+            letter='Dear {}, \n\n\tThank for for your very kind donations of ${}.\n\n\tIt will be put to very ' \
+                   'good use.\n\n\t\tSincerely, \n\n\t\t\t - The Team'.format(donor,
+                                                                                                       str(amount[len(amount)-1]))
+            file.write(letter)
+    except OSError as fileerr:
+        print(fileerr)
+        print("\nThere was an error opening the file.\n")
+        return
+    except FileNotFoundError as fnferr:
+        print(fnferr)
+        print("\nThere was an error reading the file")
+        return
+    except ValueError as verr:
+        print("There was an error writing to the file.")
+        return
+
+def func_send_thankyou():
+    ''' Send a Thank you note'''
+    print('Thank you')
+    value=0
+    name = input('Type donor name or ''list'' to see all the names=')
+    if name == 'list':
+        # List all donors if user enters 'list'
+        for donor in people.keys():
+            print(donor)
+    else:  # It is assumed that name entered in this case
+        # Prompt for donation amount
+        try:
+            value = float(input('Please enter the donation amount='))
+            # Is it in the db?
+            # If in database, append value else create a new record
+            if name in people:
+                #Add the donation
+                getlist = people[name]
+                getlist.append(value)
+        except:
+            print("Invalid value entered!")
+            return
+        else:
+            #Add the name and the donation
+            people[name] = value
```

Comment:
Something funny going on here, once I add a new donor, it breaks the code.  Continuing on and selecting option 2 or 3 kicks me out of the program.